test/pipe.js

```diff
@@ -50,23 +50,54 @@ assert.equal(shell.error(), null);
 assert.equal(result.toString(), shell.cat('resources/uniq/pipeSorted').toString());
 
 // Synchronous exec
-// TODO: add windows tests
-if (process.platform !== 'win32') {
+if (process.platform === 'win32') {
+  // Windows-specific
+
+  // Ensure the user does not have the wrong version of FIND. I.e.,
+  // require the following to return error.
+  result = shell.exec('CD..&FIND . -name no');
+  if (!shell.error()) {
+    console.error('Warning: Cannot verify piped exec: Found POSIX-like find(1) in PATH. This test assumes a Windows environment with its FIND command (fix your PATH, exit cygwin/mingw32/MSYS2).');
+  } else {
```

Comment:
Well, could just wrap it all in an IEFE. I was just imitating how the unix code was conditional on `grep(1)`, though I inverted the conditional ;-). Also, not sure how one could use an else-if ladder without repeating expressions in the subsequent conditionals. Functions with returns or the current nesting are the natural ways to do this without repetition, I think. Please let me know how I should fix this!